appium/webdriver/appium_service.py

```diff
@@ -64,62 +64,65 @@ class AppiumServiceError(RuntimeError):
+        self._cmd: Optional[List] = None
 
-    def _get_node(self):
+    def _get_node(self) -> str:
         if not hasattr(self, '_node_executable'):
             self._node_executable = find_executable('node')
         if self._node_executable is None:
             raise AppiumServiceError('NodeJS main executable cannot be found. ' +
                                      'Make sure it is installed and present in PATH')
         return self._node_executable
 
-    def _get_npm(self):
+    def _get_npm(self) -> str:
         if not hasattr(self, '_npm_executable'):
             self._npm_executable = find_executable('npm.cmd' if sys.platform == 'win32' else 'npm')
         if self._npm_executable is None:
             raise AppiumServiceError('Node Package Manager executable cannot be found. ' +
                                      'Make sure it is installed and present in PATH')
         return self._npm_executable
 
-    def _get_main_script(self):
+    def _get_main_script(self) -> str:
         if not hasattr(self, '_main_script'):
             for args in [['root', '-g'], ['root']]:
                 try:
-                    modules_root = subprocess.check_output([self._get_npm()] + args).strip().decode('utf-8')
+                    modules_root = sp.check_output([self._get_npm()] + args).strip().decode('utf-8')
                     if os.path.exists(os.path.join(modules_root, MAIN_SCRIPT_PATH)):
                         self._main_script = os.path.join(modules_root, MAIN_SCRIPT_PATH)
                         break
-                except subprocess.CalledProcessError:
+                except sp.CalledProcessError:
                     continue
             if not hasattr(self, '_main_script'):
                 try:
-                    self._main_script = subprocess.check_output(
+                    self._main_script = str(sp.check_output(
                         [self._get_node(),
                          '-e',
-                         'console.log(require.resolve("{}"))'.format(MAIN_SCRIPT_PATH)]).strip()
-                except subprocess.CalledProcessError as e:
+                         'console.log(require.resolve("{}"))'.format(MAIN_SCRIPT_PATH)]).strip())
+                except sp.CalledProcessError as e:
                     raise AppiumServiceError(e.output)
         return self._main_script
 
     @staticmethod
-    def _parse_port(args):
+    def _parse_port(args: List[str]) -> int:
         for idx, arg in enumerate(args or []):
             if arg in ('--port', '-p') and idx < len(args) - 1:
                 return int(args[idx + 1])
         return DEFAULT_PORT
 
     @staticmethod
-    def _parse_host(args):
+    def _parse_host(args: List[str]) -> str:
         for idx, arg in enumerate(args or []):
             if arg in ('--address', '-a') and idx < len(args) - 1:
                 return args[idx + 1]
         return DEFAULT_HOST
 
-    def start(self, **kwargs):
+    def start(self, **kwargs: Any) -> sp.Popen:
```

Comment:
we don't have to set kwargs type

Reply:
Hum, it has below error without `Any` .

```
appium/webdriver/appium_service.py:125: error: Function is missing a type annotation for one or more arguments
```

And I couldn't find good solution from my search.
Do you have any idea? Or can I use `type: ignore`?

[ref]
* https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

Reply:
ok, lets keep [Any](https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs) for now